source/deprecations/ember-data/v2.x.html.md

```diff
@@ -110,3 +110,85 @@ Note: The existing behavior of
 [`DS.DateTransform`](http://emberjs.com/api/data/classes/DS.DateTransform.html)
 was preserved. It is the direct usage of `Ember.Date.parse` and overriding of
 `Date.parse` that has been deprecated.
+
+#### Store.queryRecord Array Response with RESTSerializer
+
+###### until: 3.0.0
+###### id: ds.serializer.rest.queryRecord-array-response
+
+When using
+[`DS.RESTSerializer`](http://emberjs.com/api/data/classes/DS.RESTSerializer.html)
+with previous versions of Ember Data,
+[`store.queryRecord`](http://emberjs.com/api/data/classes/DS.Store.html#method_queryRecord)
+provided support for normalizing payloads containing an array of primary data.
+This behavior has been deprecated.
+
+Deprecated payload example with an array as the primary data:
+
+```js
+// GET /users?firstName="eric"
+
+{
+  "users": [{
+    "id": "1",
+    "firstName": "Eric",
+    "lastName": "Kelly"
+  }]
+}
+```
+
+Expected payload example with a single object as the primary data:
+
+```js
+// GET /users?firstName="eric"
+
+{
+  "user": {
+    "id": "1",
+    "firstName": "Eric",
+    "lastName": "Kelly"
+  }
+}
+```
+
+If you need to support an API that responds with an array as the primary data,
+you can override
+[`normalizeQueryRecordResponse`](http://emberjs.com/api/data/classes/DS.RESTSerializer.html#method_normalizeQueryRecordResponse)
+in your serializer, manipulating the payload so it matches the expected format:
```

Comment:
I think we should point out the `store.query(…).then((result) => { result.objectAt(0) })` approach before, as this might be more straightforward for people to implement.